app/services/place_hold_form/builder.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+# Responsible for building data needed to populate the place hold form.
+class PlaceHoldForm::Builder
+  def initialize(catkey:, user_token:, client:)
+    @catkey = catkey
+    @user_token = user_token
+    @client = client # SymphonyClient
+    @volumetric_calls = []
+  end
+
+  def generate
+    bib_info = Bib.new(parsed_symphony_response(:get_bib_info, @catkey, @user_token))
+    @call_list = bib_info.call_list.map { |c| Call.new record: c }
+    @holdable_locations = find_holdable_locations
+    process_volumetric_calls
+    {
+      catkey: @catkey,
+      title: bib_info.title,
+      author: bib_info.author,
+      volumetric_calls: @volumetric_calls,
+      barcode: @volumetric_calls.present? ? nil : @call_list.sample.items.sample.barcode
+    }
+  end
+
+  private
+
+    def parsed_symphony_response(symphony_call, *params)
```

Comment:
@cdmo What would you think about moving this to `symphony_client`? I think there are potential usage for it other the form builder and it is related to symphony so moving to the symphony service seems to make sense to me but want to hear your thoughts?

Reply:
I like this idea very much. I think I'd almost rather it live in it's own stand-alone class like `SymphonyClient::Parse` or `SymphonyClientParser` or something like that? Would like to handle in a separate issue though...

Reply:
yeah sure both works, I was thinking I can use that to refactor [hold_lookup](https://github.com/psu-libraries/myaccount/blob/94c2549dfe3bf4ea4dfb8085bfb55aebb1735427/app/controllers/holds_controller.rb#L176) and [item_lookup](https://github.com/psu-libraries/myaccount/blob/94c2549dfe3bf4ea4dfb8085bfb55aebb1735427/app/controllers/holds_controller.rb#L182) :)

Reply:
Cool. We can extract this method when we refactor one of those then.  There are a few other places too we could use this, like `Hold#hold_lookup`.  Not urgent by any means I think.